### PR TITLE
master: check both places to fix bug 1372618

### PR DESCRIPTION
**Pull request by sosiouxme**
https://bugzilla.redhat.com/show_bug.cgi?id=1372618
https://github.com/openshift/origin-web-console/issues/554

### CODE REVIEW

**Comment by sosiouxme** [comment]
[test]

**Comment by sosiouxme** [comment]
@openshift/api-review seems like the best fit to review this change?

**Comment by sosiouxme** [comment]
pretty sure the test failure had nothing to do with me

**Comment by sosiouxme** [comment]
[test] again for fun

**Comment by sosiouxme** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/6409/consoleFull had a different conformance error:

```
• Failure [288.339 seconds]
deploymentconfigs
/data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:767
  with revision history limits
  /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:716
    should never persist more old deployments than acceptable after being observed by the controller [Conformance] [It]
    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:715

    Expected an error to have occurred.  Got:
        <nil>: nil

    /data/src/github.com/openshift/origin/test/extended/deployments/deployments.go:696
```

[test] once more

**Comment by sosiouxme** [comment]
@csrwng PTAL

File: pkg/cmd/server/start/start_master.go [hunk 1]
```diff
@@ -527,6 +516,44 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
...
+		return nil, nil
+	}
+	if overrideConfig, err := tryOverrideConfig(oc.Options.KubernetesMasterConfig.AdmissionConfig); err != nil {
+		return nil, err
+	} else {
+		return overrideConfig, nil
+	}
+}
+
+// Look for ClusterResourceOverrideConfig in the PluginConfig
```
Comment (csrwng) [inline pkg/cmd/server/start/start_master.go]: fix godoc

File: pkg/cmd/server/start/start_master.go [hunk 2]
```diff
@@ -527,6 +516,44 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
...
+func tryOverrideConfig(ac configapi.AdmissionConfig) (*overrideapi.ClusterResourceOverrideConfig, error) {
```
Comment (csrwng) [inline pkg/cmd/server/start/start_master.go]: maybe use a more descriptive function name?

File: pkg/cmd/server/start/start_master.go [hunk 3]
```diff
@@ -527,6 +516,44 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
 	return nil
 }
 
+// Look in two potential places where ClusterResourceOverrideConfig can be specified
+func getResourceOverrideConfig(oc *origin.MasterConfig) (*overrideapi.ClusterResourceOverrideConfig, error) {
+	if overrideConfig, err := tryOverrideConfig(oc.Options.AdmissionConfig); err != nil {
```
Comment (csrwng) [inline pkg/cmd/server/start/start_master.go]: move the if to its own line, followed by 2 if's ... 
https://golang.org/doc/effective_go.html#if

File: pkg/cmd/server/start/start_master.go [hunk 4]
```diff
@@ -527,6 +516,44 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
...
+		return nil, err
+	} else if overrideConfig != nil {
+		return overrideConfig, nil
+	}
+	if oc.Options.KubernetesMasterConfig == nil { // external kube gets you a nil pointer here
+		return nil, nil
+	}
+	if overrideConfig, err := tryOverrideConfig(oc.Options.KubernetesMasterConfig.AdmissionConfig); err != nil {
```
Comment (csrwng) [inline pkg/cmd/server/start/start_master.go]: Put if in its own line, get rid of the else

File: pkg/cmd/server/start/start_master.go [hunk 5]
```diff
@@ -527,6 +516,44 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
...
+	if overridePluginConfigFile == "" {
+		return nil, nil
+	}
+	configFile, err := os.Open(overridePluginConfigFile)
+	if err != nil {
+		return nil, err
+	}
+	if overrideConfig, err := override.ReadConfig(configFile); err != nil {
+		return nil, err
+	} else {
```
Comment (csrwng) [inline pkg/cmd/server/start/start_master.go]: get rid of the else

**Comment by pweil-** [comment]
@mfojtik PTAL

**Comment by sosiouxme** [comment]
Addressed @csrwng 's feedback. Going ahead with [merge] to get it into online ASAP.

**Comment by abhgupta** [comment]
re-[test]

**Pull request merged**